### PR TITLE
Unlock SemVer range in package.json

### PR DESCRIPTION
**Pull request by leonhma**
Just a small improvement to package.json (`"extend": "^3.0.0"` instead of `"extend": "3.0.0"`) to avoid locking dependents into vulnerable versions of a package. (eg. CVE-2018-16492 in `moment >= 3.0.0, < 3.0.2`). Generally it's a good idea to let SemVer handle automatically updating dependencies. `^3.0.0` will only upgrade minor and patch versions so no breaking changes.

### CODE REVIEW

**Comment by mauriciopoppe** [comment]
Thanks for the PR, I'll move the repo to use github actions instead of travis soon

**Comment by leonhma** [comment]
The problem is not GitHub actions or anything like that. Sorry if the wording was a bit unclear.

## The Problem

Take a look at your `package.json` file: all versions are set to a single, specific version (eg. `"moment": "3.0.0"`. If another project that depends on this project cannot satisfy this version requirement (direct or transitive dependency on a different version):
1. Npm/yarn aren't able to resolve a working version
2. There is no easy way to upgrade dependencies. If because of a vulnerability I want to upgrade to `"moment": "^3.0.2"`, I can't do that because it is not allowed by the version requirement your project specifies.

## The Solution

Add a caret (`^`) in front of all version numbers.

This works because the nodejs community (for the most part, although compatibility is guaranteed) has agreed on a versioning standard. "SemVer", or "semantic versioning". The version number consists of three numbers, separated by dots (and a suffix but that's not important here): `<major>.<minor>.<patch>`. A major version is, when there are breaking changes included in the release, and dependents (projects depending on this package) would have to adjust to changes in the API. Minor versions guarantee compatibility and mean that this release contains a new feature. Patch versions just include minor Bug- and security fixes, while also preserving compatibility.

If you put a caret in front of the version numbers (eg. `^3.0.0`), It specifies "at least 3.0.0, but also minor and patch upgrades". This way bug and security fixes as well as new features, although not breaking changes are allowed.

This is exactly what this PR does. As I said, this is generally a good idea going forward and this convention seems to be used in all your other projects.

**Comment by mauriciopoppe** [comment]
I understand and sorry if I was not clear, the PR is great and to test it I'm thinking about moving the project from using travis to Github Actions

**Comment by leonhma** [comment]
Alright if you have any questions about GitHub Actions let me know.

**Comment by mauriciopoppe** [comment]
I added a Github Action workflow to the repo, please resolve the conflicts if you can

File: package.json [hunk 1]
```diff
@@ -24,16 +24,18 @@
...
-    "mr-parser": "0.2.1"
+    "extend": "^3.0.2",
+    "mr-parser": "^0.2.1"
   },
   "devDependencies": {
-    "coveralls": "2.11.4",
-    "doctoc": "0.14.2",
-    "mocha": "2.3.3",
-    "nodemon": "1.7.1",
+    "coveralls": "^2.11.4",
```
Comment (mauriciopoppe) [inline package.json:31]: I removed a few dependencies in https://github.com/mauriciopoppe/math-codegen/pull/8/files

**Comment by leonhma** [comment]
I don't even know whats wrong here. `npm install` complains about a malformatted `package.json`, but jsonlint.com says it's fine.

**Comment by mauriciopoppe** [comment]
tests passed! I see this message though:

> This branch cannot be rebased due to conflicts

I usually do this with the upstream remote being this repo:

```
git fetch upstream && git rebase upstream/master --autostash
git push origin <branch> --force
```

I think you could also squash all commits into one too

**Comment by leonhma** [comment]
Doesn't the squash and merge button on the pr page on GitHub work? It was a bit weird how vscode merged the upstream changes but this should work I think

**Comment by mauriciopoppe** [comment]
I see this:

image

**Comment by leonhma** [comment]
The first idea I have is that you could try clicking on that 'expand' arrow next to the 'rebase and merge' button and choosing squash. If that doesn't work I'll have to `git reset` or something

**Comment by mauriciopoppe** [comment]
oh you're right, I can squash and merge but not rebase and merge, thanks :)

**Pull request merged**